account_cutoff_start_end_dates/models/account_cutoff.py

```diff
@@ -0,0 +1,224 @@
+            forecast_start_date_dt = self.start_date
+            forecast_end_date_dt = self.end_date
+            if end_date_dt > forecast_end_date_dt:
+                out_days += (end_date_dt - forecast_end_date_dt).days
+            if start_date_dt < forecast_start_date_dt:
+                out_days += (forecast_start_date_dt - start_date_dt).days
+            cutoff_days = vals["total_days"] - out_days
+        else:
+            cutoff_date_dt = self.cutoff_date
+            if start_date_dt > cutoff_date_dt:
+                cutoff_days = vals["total_days"]
+            else:
+                cutoff_days = (end_date_dt - cutoff_date_dt).days
+        cutoff_amount = aml.balance * cutoff_days / vals["total_days"]
+        cutoff_amount = self.company_currency_id.round(cutoff_amount)
+
+        vals.update(
+            {
+                "cutoff_days": cutoff_days,
+                "cutoff_amount": cutoff_amount,
+            }
+        )
+
+    def get_lines(self):
+        res = super().get_lines()
+        aml_obj = self.env["account.move.line"]
+        line_obj = self.env["account.cutoff.line"]
+        if not self.source_journal_ids:
+            raise UserError(_("You should set at least one Source Journal."))
+        mapping = self._get_mapping_dict()
+        domain = [
+            ("journal_id", "in", self.source_journal_ids.ids),
+            ("display_type", "=", False),
+            ("company_id", "=", self.company_id.id),
+            ("balance", "!=", 0),
+        ]
+
+        if self.cutoff_type in ["prepaid_expense", "prepaid_revenue"]:
+            if self.forecast:
+                domain += [
+                    ("start_date", "!=", False),
+                    ("start_date", "<=", self.end_date),
+                    ("end_date", ">=", self.start_date),
+                ]
+            else:
+                domain += [
+                    ("start_date", "!=", False),
+                    ("end_date", ">", self.cutoff_date),
+                    ("date", "<=", self.cutoff_date),
+                ]
+        elif self.cutoff_type in ["accrued_expense", "accrued_revenue"]:
+            domain += [
+                ("start_date", "!=", False),
+                ("start_date", "<=", self.cutoff_date),
+                ("date", ">", self.cutoff_date),
+            ]
+        amls = aml_obj.search(domain)
+        for aml in amls:
+            line_obj.create(self._prepare_date_cutoff_line(aml, mapping))
+        return res
+
+
+class AccountCutoffLine(models.Model):
+    _inherit = "account.cutoff.line"
+
+    start_date = fields.Date(readonly=True)
+    end_date = fields.Date(readonly=True)
+    total_days = fields.Integer(readonly=True)
+    cutoff_days = fields.Integer(
+        readonly=True,
+        help="In regular mode, this is the number of days after the "
+        "cut-off date. In forecast mode, this is the number of days "
+        "between the start date and the end date.",
+    )  # Old name: prepaid_days
```

Comment:
based https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0, field name changes should be follow of a migration script, did you plan to crate any?

Reply:
@alexis-via any news about that?

Reply:
@mmequignon see https://github.com/akretion/account-closing/pull/4